aws/resource_aws_ecr_replication_configuration.go

```diff
@@ -0,0 +1,201 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsEcrReplicationConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEcrReplicationConfigurationPut,
+		Read:   resourceAwsEcrReplicationConfigurationRead,
+		Update: resourceAwsEcrReplicationConfigurationPut,
+		Delete: schema.Noop,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"registry_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"replication_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"rule": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"destination": {
+										Type:     schema.TypeList,
+										Required: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"region": {
+													Type:     schema.TypeString,
+													Required: true,
+												},
+												"registry_id": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validateAwsAccountId,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsEcrReplicationConfigurationPut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ecrconn
+
+	input := ecr.PutReplicationConfigurationInput{
+		ReplicationConfiguration: expandEcrReplicationConfigurationReplicationConfiguration(d.Get("replication_configuration").([]interface{})),
+	}
+
+	log.Printf("[DEBUG] Creating ECR Replication Configuration: %#v", input)
```

Comment:
This isn't strictly necessary, since if debug logging is enabled, the AWS SDK logs the request and response